### PR TITLE
Implement remaining `Update_Action`s for `update_database_table`.

### LINKED ISSUE #6498: Add `update_database_table` to `In-Memory` table
```
## Update an existing database table from an in-memory table or database query.

   If an update would affect mutiple rows per key then error process.
   If unmapped columns exists then an error will occur.
   
   If an existing column is missing from the source, we will insert the database
   default value for it. If this results in an error the process will fail. When
   updating a record these columns will be left unchanged.
   If `error_on_missing_columns` is True then all existing columns must be present
   of a `Missing_Column` error will occur.

   If the table doesn't exist then a `Table_Not_Found` error.
   If any of the primary key cannot be mapped a `Primary_Key_Unmatched` error.
   If column types are incompatible then a `Invalid_Value_Type` error.
Table.update_database_table
    connection:Connection
    table_name:Text
    update_action:Update_Action = Update_Action.Update_Or_Insert
    primary_key:Vector Text = Connection.get_primary_key table_name
    multi_row_update: Boolean = False
    match_columns:Match_Columns=Match_Columns.By_Name
    error_on_missing_columns:Boolean = False
    on_problems:Problem_Behavior = Problem_Behavior.Report_Warning
    -> Database_Table

type Update_Action
    ## Records are appended but if cause a primary key clash will fail.
    Insert

    ## Just update the existing records. Unmatched columns are left unchanged.
       Errors if any record is not matched in the target table.
    Update

    ## Append the records to the new table if not found.
       Updates existing records to the new values. Unmatched columns are left unchanged.
    Update_Or_Insert

    ## Appends new records, updates existing records, removes records not in the target table
    Align_Records
```

A suggestion would be to load a temp table and then do SELECT-INTO / UPDATE.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Closes #6498 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso [hunk 1]
```diff
@@ -62,3 +62,22 @@ type Query
...
+       where `c1` comes from `column_names` and `key1` comes from `key_columns`.
+    Update_From_Table (target_table_name:Text) (source_table_name:Text) (column_names : Vector Text) (key_columns : Vector Text)
+
+    ## PRIVATE
+       An SQL DELETE query that deletes from target table rows that are _not_
+       present in the source table, based on the specified key columns.
+       They key columns must be present under the same name in both tables.
+
+       This will usually be a query of the form
+       `DELETE FROM target WHERE (key_columns...) NOT IN (SELECT key_columns... FROM source)`.
```
Comment (jdunkerley) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso:82]: I wonder if a DELETE/JOIN would work better. 

Add a constant marker to the source, and then something like:

```sql
DELETE target
  FROM target 
  LEFT JOIN source ON  ...
 WHERE source.marker is NULL
```

**Review by jdunkerley — APPROVED** [review]
Looks good - just one thing to consider.

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso:82]: image
image

This kind of syntax seems to not be supported in neither of the DBs we currently have.

Comment (radeusgd) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR/Query.enso:82]: In SQLite the `DELETE` syntax only allows for a `WHERE` clause, no joins are available.
image

In Postgres, I guess we may try to play with `MERGE` but that is for a later ticket, I think - so to be revisited in #7036

**Pull request merged**